datalad/metadata/__init__.py

```diff
@@ -0,0 +1,64 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Metadata"""
+
+
+from git.config import GitConfigParser
+from os.path import join as opj
+from datalad.support.network import is_url
+
+# XXX Could become dataset method
+def get_metadata_type(ds, guess=False):
+    """Return the metadata type/scheme of a dataset
+
+    Parameters
+    ----------
+    ds : Dataset
+      Dataset instance to be inspected
+    guess : bool
+      Whether to try to auto-detect the type if no metadata type setting is
+      found. All supported metadata schemes are tested in alphanumeric order.
+      Only the label of the first matching scheme is reported.
+
+    Returns
+    -------
+    str or None
+      Metadata type label or `None` if no type setting is found and and optional
+      auto-detection yielded no results
+    """
+    cfg = GitConfigParser(opj(ds.path, '.datalad', 'config'),
```

Comment:
even though we had #437 about harmonization of config files to possibly follow git format, not yet sure if that should be the way to go... discussion will continue there...